server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -122,30 +114,64 @@ public TemporalAccessor parse(String input) {
         }
 
         try {
-            return parser.parse(input);
+            return doParse(input);
         } catch (DateTimeParseException e) {
             throw new IllegalArgumentException("failed to parse date field [" + input + "] with format [" + format + "]", e);
         }
     }
 
+    private TemporalAccessor doParse(String input) {
+        if (parsers.size() > 1) {
```

Comment:
this could be statically initialized in the ctor?

Reply:
the result of `parsers.size() > 1` ? like a flag `isCompositeParser` ? I think this is not needed (probably won't give any performance gain) maybe just a method for this expression?

Reply:
since the complexity of the size() is expected to be constant in the implementation used here, I will leave this as it is.